### ISSUE TITLE
`outdated` shows an app that is no longer installed

### ISSUE THREAD

**Issue by matteocorti** [mas-cli/mas#247]
## Your Environment

- `mas version`: 1.6.3
- macOS version (`system_profiler SPSoftwareDataType -detailLevel mini`): macOS 10.14.6 (18G95)

### mas Install Method

`brew install mas` (homebrew-core)

## Describe the Bug

`mas outdated` shows an application which cannot be uninstalled and which is not listed by `mas list`. The App Store app does not show it either.

```console
$ mas version
1.6.3
$ mas reset
$ mas list
1050307950 dDNS Broker (2.8)
768610744 Photoshop Elements 12 Editor (12.0)
425264550 Disk Speed Test (3.2)
722364523 Scan (1.0.3)
987045856 FTP Server (2.5.1)
405399194 Kindle (1.26.1)
409907375 Remote Desktop (3.9)
404221303 Citrix Receiver (11.3.2)
409183694 Keynote (9.1)
446366603 ifolor Designer (3.6.6)
682658836 GarageBand (10.3.2)
443987910 1Password (6.8.8)
409203825 Numbers (6.1)
497799835 Xcode (10.3)
408937559 iFlicks (3.0.3)
409201541 Pages (8.1)
408981434 iMovie (10.1.12)
715768417 Microsoft Remote Desktop (8.0.30030)
1014850245 Monit (2.0)
1055511498 Day One (4.1.1)
924726344 Deliveries (3.2.1)
402398561 MindNode Pro (1.11.5)
$ mas outdated
1066144116 Live Home 3D - Interior Design (unknown -> 3.7)
$ mas uninstall 1066144116
Error: Not installed
```

Attempting to upgrade makes mas hang forever.

## To Reproduce

```console
$ mas version
1.6.3
$ mas reset
$ mas list
1050307950 dDNS Broker (2.8)
768610744 Photoshop Elements 12 Editor (12.0)
425264550 Disk Speed Test (3.2)
722364523 Scan (1.0.3)
987045856 FTP Server (2.5.1)
405399194 Kindle (1.26.1)
409907375 Remote Desktop (3.9)
404221303 Citrix Receiver (11.3.2)
409183694 Keynote (9.1)
446366603 ifolor Designer (3.6.6)
682658836 GarageBand (10.3.2)
443987910 1Password (6.8.8)
409203825 Numbers (6.1)
497799835 Xcode (10.3)
408937559 iFlicks (3.0.3)
409201541 Pages (8.1)
408981434 iMovie (10.1.12)
715768417 Microsoft Remote Desktop (8.0.30030)
1014850245 Monit (2.0)
1055511498 Day One (4.1.1)
924726344 Deliveries (3.2.1)
402398561 MindNode Pro (1.11.5)
$ mas outdated
1066144116 Live Home 3D - Interior Design (unknown -> 3.7)
$ mas uninstall 1066144116
Error: Not installed
```

## Expected Behavior

`mas outdated` should not show `1066144116 Live Home 3D - Interior Design (unknown -> 3.7)`

## Actual Behavior

`mas outdated` shows `1066144116 Live Home 3D - Interior Design (unknown -> 3.7)`

## Screenshots, Terminal Output

```console
$ mas version
1.6.3
$ mas reset
$ mas list
1050307950 dDNS Broker (2.8)
768610744 Photoshop Elements 12 Editor (12.0)
425264550 Disk Speed Test (3.2)
722364523 Scan (1.0.3)
987045856 FTP Server (2.5.1)
405399194 Kindle (1.26.1)
409907375 Remote Desktop (3.9)
404221303 Citrix Receiver (11.3.2)
409183694 Keynote (9.1)
446366603 ifolor Designer (3.6.6)
682658836 GarageBand (10.3.2)
443987910 1Password (6.8.8)
409203825 Numbers (6.1)
497799835 Xcode (10.3)
408937559 iFlicks (3.0.3)
409201541 Pages (8.1)
408981434 iMovie (10.1.12)
715768417 Microsoft Remote Desktop (8.0.30030)
1014850245 Monit (2.0)
1055511498 Day One (4.1.1)
924726344 Deliveries (3.2.1)
402398561 MindNode Pro (1.11.5)
$ mas outdated
1066144116 Live Home 3D - Interior Design (unknown -> 3.7)
$ mas uninstall 1066144116
Error: Not installed
```

**Comment by paced** [comment]
+1. I'm getting this as well. If you do not have enough space on your disk and try to update XCode, for example, then uninstall XCode through `mas`, `mas outdated` will still show XCode.

**Comment by phatblat** [comment]
Should be fixed now in [1.8.1](https://github.com/mas-cli/mas/releases/tag/v1.8.1). @matteocorti can you confirm?

**Comment by matteocorti** [comment]
Sorry the machine where I could reproduce the problem was replaced two weeks ago ...

**Comment by baggiponte** [comment]
Hi! I am experiencing a similar problem. When I `mas outdated` I get a list of apps which are not outdated or not even on my machine anymore:

```
❯ mas outdated
1482920575 DuckDuckGo Privacy Essentials (1.4.3 -> 1.4.4)
1147396723 WhatsApp (2.2047.10 -> 2.2119.6)
409201541 Pages (10.3.5 -> 11.0)
1188020834 OverPicture (1.10 -> 1.12)
409203825 Numbers (10.3.5 -> 11.0)
1444383602 GoodNotes (5.6.19 -> 5.7.5)
1402042596 AdBlock (1.27.0 -> 1.28.2)
409183694 Keynote (10.3.5 -> 11.0.1)
747648890 Telegram (7.2.3 -> 7.7)
803453959 Slack (4.10.3 -> 4.16.0)
```

```
❯ mas list
1482920575  DuckDuckGo Privacy Essentials  (1.4.4)
409183694   Keynote                        (11.0.1)
1402042596  AdBlock                        (1.28.2)
803453959   Slack                          (4.16.0)
1147396723  WhatsApp                       (2.2119.6)
409201541   Pages                          (11.0)
1188020834  OverPicture                    (1.12)
747648890   Telegram                       (7.7)
1233965871  ScreenBrush                    (1.7.8)
937984704   Amphetamine                    (5.2.2)
409203825   Numbers                        (11.0)
1559269364  Notion Web Clipper             (1.0.3)
```

`mas` is `1.8.2` and System is `macOS 11.4`

Thank you!

**Comment by rgoldberg** [comment]
@baggiponte Which of the listed apps aren't out of date? All appear to have newer versions available according to the mas output. If they aren't out of date, what is the version installed on your machine, and what is the version of the app in the App Store GUI? Thanks.

**Comment by baggiponte** [comment]
Hi, I am no longer experiencing this issue!

**Comment by rgoldberg** [comment]
@baggiponte Any ideas what fixed it?

**Comment by baggiponte** [comment]
Not really sure. Perhaps simply running `mas list` was enough to update the installed app list (if I remember correctly I immediately ran `mas outdated` after installing it, before any other command).

**Comment by chris-araman** [comment]
I suspect this is resolved with the fix for #309. Apps that are copied or backed up to paths outside of `/Applications` would be listed when they shouldn't have been.
